app/models/project.rb

```diff
@@ -374,7 +374,7 @@ def homepage
   def set_dependents_count
     return if destroyed?
 
-    new_dependents_count = ActiveRecord::Base.connection.with_statement_timeout(20.minutes.to_i) do
+    new_dependents_count = ActiveRecord::Base.connection.with_statement_timeout(30.minutes.to_i) do
```

Comment:
yeah, it's called via the PackageManagerDownloadWorker whenever a Project is updated : \ Unfortunately since that codepath becomes more critical the more popular a package is, this hotspot seems worth allowing a high timeout.